### PR TITLE
fix(getToken): fix getToken()

### PR DESCRIPTION
**Pull request by parkererickson-tg**
https://dev.tigergraph.com/forum/t/tigergraphexception-cannot-request-a-token-with-username-password-for-versions-3-5/3877/8

### CODE REVIEW

**Review by qe-tigergraph — CHANGES_REQUESTED** [review]
Unit Test: SUCCESS, e2e Test: FAILURE, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/1049/

File: pyTigerGraph/pyTigerGraphAuth.py [hunk 1]
```diff
@@ -231,7 +231,7 @@ def getToken(self, secret: str = None, setToken: bool = True, lifetime: int = No
         elif not(success) and not(secret):
             res = self._post(self.restppUrl+"/requesttoken", authMode="pwd", data=str({"graph": self.graphname}), resKey="results")
             success = True
-        else:
+        elif not(success) and (int(s) < 3 or (int(s) == 3 and int(m) < 5)):
```
Comment (billshitg) [inline pyTigerGraph/pyTigerGraphAuth.py:234]: Seems like it is still missing the case when secret is given and the version is above 3.5. Do we use the secret to get token or use username/password?

**Comment by parkererickson-tg** [comment]
The case of secret given and over 3.5 should be covered in the if block below checking if the first if was success or not. This was due to the change of GET vs POST that was changed with v3.5.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by billshitg** [comment]
Is it this line: `elif not(success) and not(secret)`? If secret is given, would not(secret) be False, and this condition be False?

**Comment by parkererickson-tg** [comment]
No, it is line 238-248:
```py
if not success:
            try:
                data = {"secret": secret}

                if lifetime:
                    data["lifetime"] = str(lifetime)

                res = json.loads(requests.post(self.restppUrl + "/requesttoken",
                    data=json.dumps(data), verify=False).text)
            except Exception as e:
                raise e
```

(I don't know why it was implemented this way either)

**Review by billshitg — APPROVED** [review]

**Pull request merged**